x-pack/legacy/plugins/siem/public/containers/case/utils.ts

```diff
@@ -4,4 +4,72 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { AllCases, AllCasesSnake, Case, CaseSnake } from './types';
+
 export const getTypedPayload = <T>(a: unknown): T => a as T;
+
+export const convertCaseToCamel = (snakeCase: CaseSnake): Case => ({
```

Comment:
I was wondering if doing something generic will be better, I tried to ask doctor' TS @FrankHassanabad  if had a better way than the `as` but without the as I am getting some weird stuff in your code.
```
export const convertCaseToCamel = <T, U extends {}>(snakeCase: T): U => {
  return Object.entries(snakeCase).reduce((acc, [key, value]) => {
    if (isObject(value)) {
      set(camelCase(key), convertCaseToCamel(value), acc);
    } else {
      set(camelCase(key), value, acc);
    }
    return acc;
  }, {} as U);
};

export const convertAllCasesToCamel = (snakeCases: AllCasesSnake): AllCases => ({
  cases: snakeCases.cases.map(snakeCase => convertCaseToCamel<CaseSnake, Case>(snakeCase)),
  page: snakeCases.page,
  perPage: snakeCases.per_page,
  total: snakeCases.total,
});
```

Reply:
love that recursive function, hope we can make TS happy

Reply:
TS is happy but I am not really happy with the casting but we can always figure out the casting issue later on. I just like that it is generic.